utils/cmut__utils__compile_resource.cmake

```diff
@@ -0,0 +1,139 @@
+    file(APPEND "${FILE_HEADER}" " * @brief Name space of all auto-generateFile\n")
+    file(APPEND "${FILE_HEADER}" " */\n")
+    file(APPEND "${FILE_HEADER}" "namespace generateFile {\n")
+    file(APPEND "${FILE_HEADER}" "\n")
+    file(APPEND "${FILE_HEADER}" "/**\n")
+    file(APPEND "${FILE_HEADER}" " * @brief Data Raw of the file\n")
+    file(APPEND "${FILE_HEADER}" " */\n")
+    file(APPEND "${FILE_HEADER}" "extern std::vector<uint8_t> ${SOURCE_FILE_NAME_VARIABLE};\n")
+    file(APPEND "${FILE_HEADER}" "\n")
+    file(APPEND "${FILE_HEADER}" "/**\n")
+    file(APPEND "${FILE_HEADER}" " * @brief Worktree name of the file\n")
+    file(APPEND "${FILE_HEADER}" " */\n")
+    file(APPEND "${FILE_HEADER}" "extern std::string ${SOURCE_FILE_NAME_VARIABLE}__FileName;\n")
+    file(APPEND "${FILE_HEADER}" "\n")
+    file(APPEND "${FILE_HEADER}" "}\n")
+    file(APPEND "${FILE_HEADER}" "\n")
+    file(APPEND "${FILE_HEADER}" "#endif//${HEADER_LOCK}\n")
+    cmut_info("     ==> DONE" )
+    # Generate all the SOURCE file:
+    cmut_info(" Generate file: '${FILE_SOURCE}'" )
+    file(WRITE "${FILE_SOURCE}" "/** @file\n")
+    file(APPEND "${FILE_SOURCE}" " * @brief Include header in a C++ header\n")
+    file(APPEND "${FILE_SOURCE}" " * @copyright UNOWHY 2017\n")
+    file(APPEND "${FILE_SOURCE}" " * @licence PROPRIETARY\n")
+    file(APPEND "${FILE_SOURCE}" " * @note Auto-generated file with cmake\n")
+    file(APPEND "${FILE_SOURCE}" " */\n")
+    file(APPEND "${FILE_SOURCE}" "\n")
+    file(APPEND "${FILE_SOURCE}" "#include \"${FILE_HEADER}\"\n")
+    file(APPEND "${FILE_SOURCE}" "\n")
+    file(APPEND "${FILE_SOURCE}" "namespace generateFile {\n")
+    file(APPEND "${FILE_SOURCE}" "\n")
+    file(APPEND "${FILE_SOURCE}" "std::vector<uint8_t> ${SOURCE_FILE_NAME_VARIABLE} = {\n")
+    string(REGEX MATCHALL ".." output "${contents}")
+    string(REGEX REPLACE ";" "),\n  static_cast<uint8_t>(0x" output "${output}")
+    file(APPEND "${FILE_SOURCE}" "  static_cast<uint8_t>(0x${output})\n")
+    file(APPEND "${FILE_SOURCE}" "};\n")
+    file(APPEND "${FILE_SOURCE}" "\n")
+    file(APPEND "${FILE_SOURCE}" "std::string ${SOURCE_FILE_NAME_VARIABLE}__FileName = \"${SOURCE_FILE_NAME}\";\n")
+    file(APPEND "${FILE_SOURCE}" "\n")
+    file(APPEND "${FILE_SOURCE}" "}\n")
+    file(APPEND "${FILE_SOURCE}" "\n")
+    cmut_info("     ==> DONE" )
+endfunction()
+
+
+function(cmut__utils__generate_header_from_file HEADER_NAME SOURCE_NAME INPUT_FILE CURRENT_BINARY_DIR)
+    cmut_info(" Single file Generation : '${INPUT_FILE}'" )
+    cmut__utils__get_header_name(FILE_HEADER ${INPUT_FILE} ${CURRENT_BINARY_DIR})
+    cmut__utils__get_source_name(FILE_SOURCE ${INPUT_FILE} ${CURRENT_BINARY_DIR})
+    if(   NOT EXISTS ${FILE_HEADER}
+       OR NOT EXISTS ${FILE_SOURCE}
+       OR ${INPUT_FILE} IS_NEWER_THAN ${FILE_HEADER}
+       OR ${INPUT_FILE} IS_NEWER_THAN ${FILE_SOURCE}
+       OR ${CMAKE_CURRENT_LIST_FILE} IS_NEWER_THAN ${FILE_HEADER}
```

Comment:
not a good idee, use a version number if you really add this feature

Reply:
done